.github/workflows/ci.yaml

```diff
@@ -0,0 +1,116 @@
+
+    runs-on: windows-2019
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v2.3.3
+
+    - name: setup environment - static build
+      if: matrix.build_variant == 'static'
+      run: |
+        echo "::set-env name=LIBT_STATIC_RT::ON"
+        echo "::set-env name=LIBT_BSL::OFF"
+        echo "::set-env name=VCPKG_TRIPLET::x64-windows-static"
+        echo "::set-env name=BUILD_VARIANT::static"
+
+    - name: setup environment - build with no deprecated functions
+      if: matrix.deprecated_functions == 'no_depr_fun'
+      run: |
+        echo "::set-env name=LIBT_TESTS::OFF"
+        echo "::set-env name=LIBT_DEPR_FUN::OFF"
+
+    # NOTE: MSVC tools must be in the path for the Ninja generator. Caveat: must use cmd
+    - name: setup MSVC dev cmd
+      uses: ilammy/msvc-dev-cmd@v1.4.1
+
+    # ninja is preferrable, not a hard requirement
+    - name: install additional required packages with chocolatey
+      run: |
+        choco install ninja
+
+    - name: setup vcpkg (cached, if possible)
+      uses: lukka/run-vcpkg@v4.0
+      with:
+        vcpkgDirectory: ${{ env.VCPKG_DEST_WIN }}
+        vcpkgGitCommitId: ${{ env.VCPKG_COMMIT }}
+        setupOnly: true
+        appendedCacheKey: ${{ matrix.build_variant }}
+
+    # clear buildtrees to reduce disk space requirements
+    - name: install dependencies via vcpkg
+      run: |
+        ${{ env.RUNVCPKG_VCPKG_ROOT }}/vcpkg.exe install `
+          boost-asio:${{ env.VCPKG_TRIPLET }} `
+          boost-chrono:${{ env.VCPKG_TRIPLET }} `
+          boost-config:${{ env.VCPKG_TRIPLET }} `
+          boost-crc:${{ env.VCPKG_TRIPLET }} `
+          boost-date-time:${{ env.VCPKG_TRIPLET }} `
+          boost-iterator:${{ env.VCPKG_TRIPLET }} `
+          boost-multiprecision:${{ env.VCPKG_TRIPLET }} `
+          boost-pool:${{ env.VCPKG_TRIPLET }} `
+          boost-random:${{ env.VCPKG_TRIPLET }} `
+          boost-scope-exit:${{ env.VCPKG_TRIPLET }} `
+          boost-system:${{ env.VCPKG_TRIPLET }} `
+          boost-variant:${{ env.VCPKG_TRIPLET }} `
+          openssl:${{ env.VCPKG_TRIPLET }} `
+          --clean-after-build
+
+    - name: build libtorrent
+      shell: cmd
+      run: |
+        cmake -B cmake-build-dir -G "Ninja" ^
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_config }} ^
+          -DCMAKE_TOOLCHAIN_FILE=${{ env.VCPKG_DEST_WIN }}\scripts\buildsystems\vcpkg.cmake ^
+          -DVCPKG_TARGET_TRIPLET=${{ env.VCPKG_TRIPLET }} ^
+          -Dbuild_examples=ON -Dbuild_tools=ON ^
+          -Dbuild_tests=${{ env.LIBT_TESTS }} -Ddeprecated-functions=${{ env.LIBT_DEPR_FUN }} ^
+          -DBUILD_SHARED_LIBS=${{ env.LIBT_BSL }} -Dstatic_runtime=${{ env.LIBT_STATIC_RT }} ^
+          --graphviz=cmake-build-dir\target_graph.dot
+        cmake --build cmake-build-dir
+
+    - name: run libtorrent tests
+      run: |
+        ctest
+
+    - name: upload artifact as zip
+      uses: actions/upload-artifact@v2.2.0
```

Comment:
what does this do? where does it upload to?

Reply:
@arvidn 

> where does it upload to?

"Somewhere on GitHub's servers". The storage is unlimited for open source projects as far as I know. You can see them in the page (you can download the artifacts by clicking on them): https://github.com/FranciscoPombal/libtorrent/actions/runs/287100024

Reply:
Also: https://github.com/actions/upload-artifact#where-does-the-upload-go

Reply:
I think I would prefer to upload less. either nothing, or just some select binaries, like the examples and main shared library perhaps